index.bs

```diff
@@ -12995,207 +12873,279 @@ with the following characteristics:
+<h5 id="es-map-iterator">@@iterator</h5>
 
-An <code class="idl">entries</code> data property must exist on
+There must exist a data property whose name is the {{@@iterator}} symbol on
 |A|’s [=interface prototype object=]
 with attributes { \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
 and whose value is the [=function object=] that is the value of
-the {{@@iterator}} property.
+the <code>entries</code> property.
+
+<div algorithm>
+    To <dfn>create a map iterator</dfn>
+    from a [=/map=] |map|
+    and a |kind| ("key+value", "key", or "value"):
+
+    1. Let |closure| be a new [=/Abstract Closure=] with no parameters
+        that captures |map| and |kind| and performs the following steps when called:
+        1. [=map/For each=] |key|→|value| of |map|'s [=map entries=]:
+            1. Set |key| and |value| to each [=converted to an ECMAScript value=].
+            1. If |kind| is "key", let |result| be |key|.
+            1. Else if |kind| is "value", let |result| be |value|.
+            1. Else, let |result| be [$CreateArrayFromList$](« |key|, |value| »)
+            1. Perform [=?=] [$GeneratorYield$]([$CreateIterResultObject$](|result|, <emu-value>false</emu-value>)).
+
+            Note: The [=map/size=] of |map|, and the order of its entries,
+            might have changed while execution of this abstract operation
+            was paused by [$Yield$].
+        1. Return <emu-value>undefined</emu>.
+    1. Return [$CreateIteratorFromClosure$](|closure|, "%MapIteratorPrototype%", {{%MapIteratorPrototype%}}).
+</div>
 
 
-<h5 id="es-map-keys-values">keys and values</h5>
+<h5 id="es-map-entries">entries</h5>
 
-For both of <code class="idl">keys</code> and <code class="idl">values</code>, there must exist a data property with that name on
+There must exist an <code class="idl">entries</code> data property on
 |A|’s [=interface prototype object=]
 with the following characteristics:
 
 *   The property has attributes { \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }.
-*   The value of the property is a [=built-in function object=] that [=forwards to the internal map object|forwards that name to the internal map object=].
+*   The value of the property is a [=built-in function object=]
+    whose behavior when invoked is as follows:
+
+    <div algorithm="to invoke the entries method of Maps">
+        1. Let |O| be the <emu-val>this</emu-val> value, [=implementation-checked=] against <var ignore>a</var> with identifier "<code>entries</code>" and type "<code>method</code>".
+        1. Let |map| be |O|'s [=map entries=].
+        1. Return the result of [=creating a map iterator=] from |map| with kind "key+value".
+    </div>
 
 The value of the [=function objects=]’ <code class="idl">length</code> properties is the Number value <emu-val>0</emu-val>.
```

Comment:
Here and elsewhere, it should be

```suggestion
The value of the [=function object=]'s <code class="idl">length</code> property is the Number value <emu-val>0</emu-val>.
```

This unfortunately got copied into a bunch of other places instead of the other places' singular version getting copied here :)